JavaScript-2 Assignments/Question 2,3,4/js/singlelinkedlist.js

```diff
@@ -0,0 +1,253 @@
+            start = start.next;
+        }
+        start.next = newNode;
+        this.length++;
+    }
+}
+
+/**
+ * display the linked list
+ * @return result - either containing Linked List or empty
+ */
+SinglyLinkedList.prototype.display = function () {
+    var start = this.head;
+    var result = "";
+    if (start === null) {
+        result = "Empty";
+    } else {
+        while (start != null) {
+            result += start.data + "-->";
+            start = start.next;
+        }
+        result += null;
+    }
+
+    return result;
+}
+
+/**
+ * add the element into linked list according to the position given
+ * @param {*} data - input data
+ * @param {*} pos - positon at which data inserted
+ * @return msg - message according to conditions
+ */
+SinglyLinkedList.prototype.addAtPosition = function (data, pos) {
+    var start = this.head;
+    var count = 1;
+    var newNode = new Node(data);
+    var maxLength = this.length + 1;
+    var msg = "";
+    if (pos > maxLength || pos < 1) {
+        msg = "Invalid Position";
+    }
+    else if (pos == 1) {
+        newNode.next = this.head;
+        this.head = newNode;
+        this.length++;
+        msg = "Added Successfully at position 1";
+    }
+    else {
+        while (count != pos - 1) {
+            start = start.next;
+            count++;
+        }
+        newNode.next = start.next;
+        start.next = newNode;
+        this.length++;
+        msg = "Added at position " + (count + 1);
+    }
+
+    return msg;
+}
+
+/**
+ * Return the value/data by its position
+ * @param {*} position - position of data
+ * @return msg - message 
+ */
+SinglyLinkedList.prototype.getValueByPosition = function (position) {
+    var start = this.head;
+    var msg;
+    var maxLength = this.length;
+    var count = 1;
+    if (start === null) {
+        msg = "Empty Linked List";
+    }
+    else if (position > maxLength || position < 1) {
+        msg = "Invalid Position";
+    }
+    else {
+        while (count != position) {
+            start = start.next;
+            count++;
+        }
+        msg = start.data;
+    }
+
+    return msg;
+}
+
+/**
+ * This function check input value is empty or Not
+ * @param {*} value - input value
+ * @return boolean - true if valid else false
+ */
+function checkInput(value) {
+    var flag = true;
+    if (value === "") {
+        flag = false
+    }
+
+    return flag;
+}
+
+/**
+ * Display Message
+ * @param {*} message - message to display
+ */
+function displayMessage(message) {
+    document.getElementById("message").innerHTML = message;
+}
+
+/**
+ * Display the Linked List
+ */
+function displayLinkedList() {
+    document.getElementById("displayLinkedList").innerHTML = obj.display();
+}
+
+/**
+ * Clear the input field
+ * @param {*} input 
+ */
+function clearInputField(input) {
+    input.value = "";
+}
+
+/**
+ * Get value from prompt and check 
+ * @return position - value fetch from prompt
+ */
+function getPromptData() {
```

Comment:
This is the same function used in above program, this can be moved in a common file.